misc-utils/rename.c

```diff
@@ -77,18 +77,21 @@ static int do_symlink(char *from, char *to, char *s, int verbose, int noact, int
 	if (string_replace(from, to, target, target, &newname))
 		ret = 0;
 
-	if (ret == 1 && nooverwrite && lstat(target, &sb) == 0) {
-		if (verbose)
-			printf(_("Skipping existing link: `%s' -> `%s'\n"), s, target);
-
-		ret = 0;
-	}
-
 	if (ret == 1) {
-		if (!noact && 0 > unlink(s)) {
+		if (access(s, F_OK) != 0) {
+			warn(_("%s: not accessible"), s);
```

Comment:
I'm not sure with access() here, it fails if the symlink target does not exist. It makes rename command useless to modify  broken symlinks. See our regression tests
```
cd tests
./run.sh rename
```
the test ts/rename/subdir contains broken symlinks.

Reply:
FIxed by calling faccessat().

Also placed this call before lstat() so that it has a chance to fail.